File: RentApp/ClientApp/src/app/models/reserv.ts
import { ClientReserv } from './client-reserv';

export interface Reserv {
    id: number;
    checkIn: Date;
    checkOut: Date;
    realPayment: number;
    fictionalPayment: number;
    reservationMethod: string;
    consume?: boolean;
    rating: number;

    clientsReserv: ClientReserv[];
}

File: RentApp/Models/Reserv.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace RentApp.Models
{
    public class Reserv : IEntity
    {
        public int Id { get; set; }
        public DateTime CheckIn { get; set; }
        public DateTime CheckOut { get; set; }
        public float RealPayment { get; set; }
        public float FictionalPayment { get; set; }
        public string ReservationMethod { get; set; }
        public bool? Consume { get; set; }
        public int Rating { get; set; }

        public List<ClientReserv> ClientsReserv { get; set; }
    }
}
